library/build.go

```diff
@@ -138,7 +139,7 @@ func (b *Build) Environment() map[string]string {
 	// check if the Build event is tag
 	if strings.EqualFold(b.GetEvent(), constants.EventTag) {
 		// capture the tag reference
-		tag := ToString(strings.TrimPrefix(b.GetRef(), "refs/tags/"))
+		tag := ToString(strings.SplitN(b.GetRef(), "refs/tags/", 2)[1])
```

Comment:
Match the existing environment variable injected via the compiler https://github.com/go-vela/compiler/blob/master/compiler/native/environment.go#L225.